cds_ils/cli.py

```diff
@@ -198,3 +241,290 @@ def run_command(command, catch_exceptions=False, verbose=True):
+
+
+def mint_record_pid(pid_type, pid_field, record):
+    """Mint the given PID for the given record."""
+    PersistentIdentifier.create(
+        pid_type=pid_type,
+        pid_value=record[pid_field],
+        object_type="rec",
+        object_uuid=record.id,
+        status=PIDStatus.REGISTERED,
+    )
+    db.session.commit()
+
+
+@demo.command()
+@click.option("--json-path", "docs_path")
+@with_appcontext
+def create_demo_docs(docs_path):
+    """Insert real demo docs."""
+    indexer = RecordIndexer()
+
+    demo_docs = os.path.join(CURRENT_DIR, docs_path)
+    with open(os.path.join(demo_docs)) as f:
+        demo_data = json.loads(f.read())
+
+    for d in demo_data:
+        doc = Document.create(d)
+        mint_record_pid(DOCUMENT_PID_TYPE, "pid", doc)
+        doc.commit()
+        db.session.commit()
+        indexer.index(doc)
+
+    click.secho("Docs were created successfully.", fg="blue")
+
+
+def get_pids(search):
+    """Get a list of pids of the search results."""
+    s = search()
+    results = s.execute()
+    pids = []
+    for hit in results.hits:
+        pids.append(hit["pid"])
+
+    return pids
+
+
+@demo.command()
+@click.option("--json-path", "items_path")
+@with_appcontext
+def create_demo_items(items_path):
+    """Insert real demo items."""
+    indexer = RecordIndexer()
+
+    demo_items = os.path.join(CURRENT_DIR, items_path)
+    with open(os.path.join(demo_items)) as f:
+        demo_items = json.loads(f.read())
+
+    intloc_pids = get_pids(InternalLocationSearch)
+
+    for i in demo_items:
+        i["internal_location_pid"] = random.choice(intloc_pids)
+        item = Item.create(i)
+        mint_record_pid(ITEM_PID_TYPE, "pid", item)
+        item.commit()
+        db.session.commit()
+        indexer.index(item)
+
+    click.secho("Items were created successfully.", fg="blue")
+
+
+@demo.command()
+@click.option("--user-email", "user_email")
+@click.option("--is-past-loan", is_flag=True)
+@with_appcontext
+def create_loan(user_email, is_past_loan):
+    """Create a loan."""
+    patron = User.query.filter_by(email=user_email).one_or_none()
+    patron_pid = patron.get_id()
+
+    loc_pid, _ = current_app_ils.get_default_location_pid
+
+    librarian_pid = (
+        User.query.filter_by(email="librarian@test.ch").one_or_none().get_id()
+    )
+
+    delivery = list(
+        current_app.config["ILS_CIRCULATION_DELIVERY_METHODS"].keys()
+    )[randint(0, 1)]
+
+    loan_dict = {
+        "pid": RecordIdProviderV2.create().pid.pid_value,
+        "patron_pid": "{}".format(patron_pid),
+        "pickup_location_pid": "{}".format(loc_pid),
+        "transaction_location_pid": "{}".format(loc_pid),
+        "transaction_user_pid": "{}".format(librarian_pid),
+        "delivery": {"method": delivery},
+    }
+
+    if is_past_loan:
+        loan_dict["state"] = "ITEM_RETURNED"
+        loan_dict["document_pid"] = "qaywb-gfe4B"
+        transaction_date = start_date = arrow.utcnow() - timedelta(days=365)
```

Comment:
where is `start_date` defined?

Reply:
In the same line with `transaction_date`, I assign the same value to both of them.